src/Language/PureScript/Sugar/ObjectWildcards.hs

```diff
@@ -45,9 +45,9 @@ desugarObjectConstructors (Module ss coms mn ds exts) = Module ss coms mn <$> ma
     obj <- freshIdent'
     Abs (Left obj) <$> wrapLambda (ObjectUpdate (argToExpr obj)) ps
   desugarExpr (ObjectUpdate obj ps) = wrapLambda (ObjectUpdate obj) ps
-  desugarExpr (Accessor prop u) | isAnonymousArgument u = do
+  desugarExpr (Accessor prop u) | isAnonymousAccessorChain u = do
```

Comment:
Instead of doing this in two passes, why not collect the property names on the way down?

```haskell
peelAccessorChain :: Expr -> Maybe [String]
```

at which point you can use a fold in place of `buildAccessor`, and pattern guard here:

```haskell
desugarExpr (Accessor prop u)
  | Just props <- peelAccessorChain u = ...
```